tutorbackup/templates/backup/build/backup/backup_services.py

```diff
@@ -51,8 +53,30 @@ def mysqldump():
         logger.info(f"Dumping MySQL databases {mysql_databases} "
                     f"on {host}:{port} to {outfile}")
     else:
-        databases_statement = "--all-databases"
-        logger.info(f"Dumping all MySQL databases "
+       dblist_file = MYSQL_DBS_LIST_FILE
+        sql_query="""SELECT schema_name FROM information_schema.schemata WHERE schema_name NOT IN
+             ('mysql', 'sys', 'performance_schema', 'information_schema')"""
+
+        mysql_cmd = ("mysql "
+            f"--host={host} --port={port} "
+            f"--user={user} --password={password}"
+            f""" -ANe"{sql_query}" > {dblist_file}""")
+
+        logger.info("Creating file with the list of database schemas to backup")
+        check_call(mysql_cmd,
+                shell=True,
+                stdout=sys.stdout,
+                stderr=sys.stderr)
+
+        logger.info("Reading the list of databases to backup")
+        databases = ""
+        with open(dblist_file, 'r') as dblist:
+            for line in dblist:
+                databases = databases+line.strip()+" "
+
+#        databases_statement = "--all-databases"
+        databases_statement = f"--databases {databases}"
+        logger.info(f"Dumping MySQL databases "
```

Comment:
I'm still trying to understand why we need the detour via the text file. Isn't this simpler to do by using `mysql -B`, and `check_output` in combination with `.splitlines()`, instead of `check_call` and a separate file?

```suggestion
        sql_query="""SELECT schema_name FROM information_schema.schemata WHERE schema_name NOT IN
             ('mysql', 'sys', 'performance_schema', 'information_schema')"""

        mysql_cmd = ("mysql "
            f"--host={host} --port={port} "
            f"--user={user} --password={password} "
            f"""-BNe "{sql_query}"""")

        logger.info("Retrieving the list of database schemas to backup")
        databases = check_output(mysql_cmd,
                                 shell=True,
                                 text=True).splitlines()

        databases_statement = f"--databases {databases}"
        logger.info(f"Dumping MySQL databases "
```

Maybe I'm missing something though. Can you please explain your approach?

Reply:
You are right, this approach would probably be more efficient and there is no need for a separate file with the list of the schemas. I just tried the first thing that came in mind (I had tried a separate file in some other cases where this file could be used elsewhere as well, so this is what I was used to do, but here you are right, it's not necessary).
I will make a new commit with your approach and let you know.

Reply:
Ok I have created a new commit with the changes, please let me know if this seems a bit better!